components/url/origin.rs

```diff
@@ -101,7 +101,7 @@ pub struct OpaqueOrigin(Uuid);
 known_heap_size!(0, OpaqueOrigin);
 
 /// A representation of an [origin](https://html.spec.whatwg.org/multipage/#origin-2).
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
That is, the origin of a request should be a snapshot of the global's origin at the moment it was initiated.